### PR TITLE
Add crypt4gh 1.0

### PR DESCRIPTION
**Pull request by blankdots**
### Describe the pull request:

- [x] Functional change
- [x] New feature
- [x] Build system change

### Pull request long description:

Update LocalEGA with crypt4GH 1.0 using upstream changes from EGA-archive/LocalEGA. At the same time move from Travis CI to Github Actions.

### Changes made:

1. Added upstream changes for crypt4GH 1.0 
2. Added actions for style check, unit tests (both python 3.6. and 3.7) and integration tests
3. Removed travis integration
4. configurable path for `verify.py` that was lost in the cherry-picking
5. fix test fro `wrong-fileformat.bats`

### Related issues:

`integrate changes from EGA-arhive/LocalEGA` card on the Tryggve2 SDA board

### Additional information:

Tried to make the change to Apache Mina for #10 however that will require more changes eventually the inclusion of https://github.com/neicnordic/LocalEGA-deploy-init and I think those should be tackled in a different PR.
Did not test the Cucumber tests, as not sure they are up to date, also will be tested as part of another PR, as they are also commented out on `master` branch.

### Documentation change:
None so far, but is needed.

### CODE REVIEW

File: .github/workflows/unit.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      run: sudo apt-get install libpq-dev
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox tox-gh-actions
+    - name: Run unit tests
+      env:
+        COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+      run: tox -e unit_tests
+
```
Comment (jbygdell) [inline .github/workflows/unit.yml]: trailing whitespace

File: .github/workflows/stytle.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install postgresql-devel
+      run: sudo apt-get install libpq-dev
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox tox-gh-actions
+    - name: Test with tox
+      run: tox -e flake8
```
Comment (jbygdell) [inline .github/workflows/stytle.yml]: Missing new line at EOF.

File: deploy/bootstrap/users.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    do
+	echomsg "\t\t - User ${user}"
+	passphrase=$(generate_password 8)
+	PASSPHRASES[$user]=$passphrase
+	rm -rf $user
+	ssh-keygen -t ed25519 -f ${USERS_DIR}/$user -N "$passphrase" -C "$user"@LocalEGA &>/dev/null
+	
+	# Bcrypt hash
+	passphrase_hash=$(echo -n $passphrase | \
+	${PYTHONEXEC} -c 'import bcrypt; import sys; sys.stdout.buffer.write(bcrypt.hashpw(sys.stdin.buffer.read(), bcrypt.gensalt()))')
```
Comment (jbygdell) [inline deploy/bootstrap/users.sh:34]: Not really a fan of these long oneliners.

File: deploy/bootstrap/users.sh [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    USERS_DIR=$(pwd)/users # includes expansion
+
+    rm -rf ${USERS_DIR}
+    mkdir -p ${USERS_DIR}
+
+    declare -A USERS=([dummy]=15001 [john]=15002 [jane]=15003)
+    declare -A PASSPHRASES
+
+    # check that bcrypt is installed for this python version
+    if ! ${PYTHONEXEC} -c 'from importlib.util import find_spec; import sys; sys.exit(0 if find_spec("bcrypt") is not None else 1)'; then
```
Comment (jbygdell) [inline deploy/bootstrap/users.sh:18]: This should be able do be done in a prettier way than a combined bash python oneliner.

**Review by jbygdell — CHANGES_REQUESTED** [review]
There is a bunch of very long one-liners that should be able to be done i a cleaner manner.

File: tests/_others/simple.bats [hunk 5]
```diff
@@ -33,19 +52,24 @@ function teardown() {
...
+
+    [ -n "${TESTUSER}" ]
+    [ -n "${TESTUSER_SECKEY}" ]
+    [ -n "${TESTUSER_PASSPHRASE}" ]
 
     # Generate a random file (1 MB)
-    legarun c4gh_generate 1 /dev/urandom ${TESTFILES}/${TESTFILE}
-    [ "$status" -eq 0 ]
+    export C4GH_PASSPHRASE=${TESTUSER_PASSPHRASE}
+    dd if=/dev/urandom bs=1048576 count=1 | crypt4gh encrypt --sk ${TESTUSER_SECKEY} --recipient_pk ${EGA_PUBKEY} > ${TESTFILES}/${TESTFILE}.c4ga
```
Comment (jbygdell) [inline tests/_others/simple.bats:63]: to long one-liner

File: tests/integration/inbox.bats [hunk 6]
```diff
@@ -73,7 +88,8 @@ function teardown() {
     [ "$status" -eq 0 ]
 
     # Encrypt it in the Crypt4GH format
-    legarun lega-cryptor encrypt --pk ${EGA_PUB_KEY} -i ${TESTFILES}/${TESTFILE} -o ${TESTFILES}/${TESTFILE}.c4ga
+    export C4GH_PASSPHRASE=${TESTUSER_PASSPHRASE}
+    crypt4gh encrypt --sk ${TESTUSER_SECKEY} --recipient_pk ${EGA_PUBKEY} < ${TESTFILES}/${TESTFILE} > ${TESTFILES}/${TESTFILE}.c4ga
```
Comment (jbygdell) [inline tests/integration/inbox.bats:92]: To long one-liner

File: tests/integration/ingestion.bats [hunk 7]
```diff
@@ -25,22 +36,30 @@ function setup() {
...
 
+    [ -n "${TESTUSER}" ]
+    [ -n "${TESTUSER_SECKEY}" ]
+    [ -n "${TESTUSER_PASSPHRASE}" ]
+
     # Generate a random file
-    legarun c4gh_generate $size $inputsource ${TESTFILES}/${TESTFILE}
-    [ "$status" -eq 0 ]
+    export C4GH_PASSPHRASE=${TESTUSER_PASSPHRASE}
+    dd if=$inputsource bs=1048576 count=$size | crypt4gh encrypt --sk ${TESTUSER_SECKEY} --recipient_pk ${EGA_PUBKEY} > ${TESTFILES}/${TESTFILE}.c4ga
```
Comment (jbygdell) [inline tests/integration/ingestion.bats:57]: to long one-liner

**Review by jbygdell — CHANGES_REQUESTED** [review]

**Comment by blankdots** [comment]
will address most of the comments in #18

**Review by viklund — APPROVED** [review]
I've already reviewed this in EGA-archive.

**Review by jbygdell — APPROVED** [review]

**Pull request merged**